### DISCUSSION TITLE
Grain orientation for the calculations of Schmid factors and Taylor factors

### DISCUSSION THREAD

**Discussion by johnyejungu** [mtex-toolbox/mtex#1077] [Ask Anything] [answered]
In the command SchmidFactor, it is for the loading direction defined as the grain orientation. While in the command calcTaylor, the input of the grain orientation is taking inverse, e.g. calcTaylor(inv(oriGrid)*epsilon,sS.symmetrise). My understanding is they are actually the same grain orientation. Original and inversevse grain orientation in the two commands come from the different definitions/equations of Schmid factors and Tayor factors. Can anyone help confirm it? Thanks.

**Comment by ralfHielscher** [comment, marked as answer]
Hi John,

the command ```SchmidFactor``` requires as input the slip systems and the stress tensor, while ```calcTaylor``` requires the slip systems and the strain tensor. The orientation comes into play only indirectly as slips systems and stress / strain tensor have to be in the same reference system. It does not matter whether this is the crystal or specimen reference system.

Ralf.

**Reply by johnyejungu** [reply]
Thanks for the clarification, Ralf. This is exactly what I am thinking about. My understanding is for the grain with an orientation Ori, if I want to calculate the Schmid Factor and Taylor factor at the same time, I can use the command "SchmidFator" with Ori and use the command "calcTaylor" with inv(Ori). Am I right?

**Reply by hsheng94** [reply]
Hi, Yejun. I think in MTEX, to apply Ori is to transform the orientation from crystal to sample reference frame and inv(ori) to transform from sample to crystal. For example, in the Taylor model example,  'calcTaylor(inv(grains.meanOrientation).*eps,sS)' use inv(ori) to transform the strain tensor  in sample reference frame to crystal reference frame.  You can also do 'calcTaylor(eps,grains.meanOrientation.*sS)' to transform slip systems in crystal reference frame to sample reference frame.

**Reply by johnyejungu** [reply]
Thanks for the confirmation! Now I'm confident I'm understanding about the orientations in MTEX correctly :)
